modules/internal/ChapelArray.chpl

```diff
@@ -4405,6 +4409,12 @@ module ChapelArray {
   // Swap operator for arrays
   //
   inline proc <=>(x: [?xD], y: [?yD]) {
+    if x.rank != y.rank then
+      compilerError("rank mismatch in array swap");
+
+    if boundsChecking then
+      checkArrayShapesUponAssignment(x, y, forSwap=true);
```

Comment:
Thanks for noticing and fixing this case!

If for any reason you preferred, I think it would be fine not to specialize the halt for swapping vs. assigning in the sense that I think it'd be reasonable to consider a swap a form of assignment (i.e., I wouldn't expect a user to be confused if their `A <=> B` statement talked in terms of assignment.

Reply:
Yeah I agree that that wouldn't be a big deal. But this is more accurate and can be more helpful I think.